tensorflow/java/src/gen/resources/tftypes.csv

```diff
@@ -0,0 +1,21 @@
+TF type,datatype code,Java type,Java boxed type,Creator?,default,Description
+Float,1,float,Float,y,0f,32-bit single precision floating point number
+Double,2,double,Double,y,0,64-bit double precision floating point number
+Int32,3,int,Integer,y,0,32-bit signed integer
+UInt8,4,byte,Byte,n,,8-bit unsigned integer
+Int16,5,,Short,n,(short)0,16-bit signed integer

```

Comment:
I am happy to remove it. I guess we were seeing the role of this file a little differently. You are looking at it as something you would like to minimize as much as possible. I was viewing it as a kind of very-lightweight spec, analogous to the proto buf specs, that summarizes how the Java API handles TF data types. A little redundancy in a spec can be good because it allows tools and humans to check for errors, though I still actually think DataType ought to be generated from it so there aren't two specs.

By the way, DataType is currently only visible in the API because Tensor lets you create a Tensor by giving an array of longs and a DataType, because you can ask what DataType a Tensor has, and because the API talks about types in terms of it. Once the ops are expressed in terms of the types package, DataType will become almost completely an implementation detail that is present mainly for backward compatibility. Probably all the javadoc specs should be written in terms of the types package once that happens.